src/components/TodoList/TodoList.js

```diff
@@ -1,41 +1,69 @@
 import React from 'react';
 import './TodoList.css';
 
-function handleClick() {
-  console.log(this)
-}
-
 class TodoList extends React.Component {
 
-  // state = {
-  //   list: [
-  //     { id: 1, name: 'Fumer des cannes', active: true },
-  //     { id: 2, name: 'Coder cette shit', active: true },
-  //     { id: 3, name: 'Nager avec des baleines', active: true },
-  //     { id: 4, name: 'Rouler des cannes pour demain', active: true }
-  //   ],
-  //   newTaskName: "",
-  // // }
-  //
-  // handleClick = () => {
-  //   console.log(this)
-  // }
+  state = {
+    list: [ ],
+    newTaskName: "",
+    editTaskName: ""
```

Comment:
So here, having `editTaskName` in the TodoList general state and not inside each of the tasks means that the `editTaskName` will be share by all tasks. Try to edit a task, fill the input and then before saving, click on "edit" of another task. What happens ?

Reply:
Is it what you expected ? What can you do to solve this issue ? Note that it really depends on what UX you expect before taking any decision. For instance, you could decide that only one task is editable at a time. Also, when you edit a task, you would maybe expect to have the input already filled with the name of task. 
Think about what you want as a result first, and then determine how you implement it. Of course, we will discuss it together for now !